docs/user-documentation/context.md

```diff
@@ -0,0 +1,33 @@
+## Context in Islandora 8
+
+[Context](https://www.drupal.org/project/context) is a Drupal contrib module that allows configuration of "if this, then that" logic using an administrative user interface. Context defines "conditions" and "reactions" to enact the "if this [condition], then that [reaction]" logic. Some Islandora 7.x repositories use the community-contributed [Islandora Context](https://github.com/SFULibrary/islandora_context) module to apply this sort of logic to Islandora objects. An example Context from Islandora 7.x is
+
+> If an object's namespace is 'customnamespace', render the block containing the rights statement "Please contact Special Collections before reusing this item".
+
+The Context project page on drupal.org says "You can think of each context as representing a 'section' of your site", but that doesn't tell the whole story, since it gives the impression that Context is all about defining subsites. Context in Drupal 8 is much more powerful in Drupal 8 than in Drupal 7 due to how it interacts with the rest of Drupal (specifically, through Drupal 8's new plugin API). Because of this increased flexibility and power, and because Context provides a ready-made user interface for site administrators to combine conditions and reactions, Islandora 8 uses Context to drive the logic underlying many important aspects of Islandora, including which derivatives get generated and how objects are displayed. In Islandora 7, many of these things were managed (and often hard-coded) within solution packs and utility modules.
+
+### An Example
+
+Let's look at the example of how Context can determine how an object is diplayed. Drupal 8 has the idea of "view modes", which allow site builders to choose what happens when an object is viewed by the user (it has nothing to do with Drupal Views). In the node edit form for Islandora objects, there is a checkbox that, if checked, tells Drupal to render the image using the OpenSeadragon viewer:
+
+![Display Hints field in node edit form](../assets/context_display_hints.png)
+
+ This  functionality is accomplished via the "Open Seadragon" Context, which, as its configuration suggests, checks as its condition whether the node as the "Open Seadragon" tag and if so, reacts by using the view mode "Open Seadragon":
+
+![Open Seadragon Context configuration](../assets/context_openseadragon_configuration.png)
+
+### Context and Actions
```

Comment:
May want to discuss how custom actions can be defined.  Most of the actions used by Islandora is defined using Plugins.

Reply:
Thanks @Natkeeran, I'll do that this evening.

Reply:
@Natkeeran I've added an example of creating an Action that thanks the user when they create a Media and tag it as a "Preservation Master File." Check it out: https://github.com/mjordan/CLAW/blob/mjordandocsprint/docs/user-documentation/context.md.